drivers/ipm/ipm_adsp.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2019, Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <device.h>
+#include <soc.h>
+#include <drivers/ipm.h>
+
+#include <platform/mailbox.h>
+#include <platform/shim.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ipm_adsp, CONFIG_IPM_LOG_LEVEL);
+
+/* FIXME define in DTS */
+#define DT_IPM_ADSP_0_BASE_ADDRESS	0x10
+#define DT_IPM_ADSP_0_NAME		"IPM_0"
+#define DT_IPM_ADSP_0_IRQ		SOC_AGGREGATE_IRQ(7, 6)
```

Comment:
@dcpleung Do you think we need to put theses parameters to Device Tree? How can I use `SOC_AGGREGATE_IRQ(7, 6)` there?

Reply:
You can looking the dtsi file for `intel_s1000_crb`. The DMA controllers there are under the CAVS interrupt contoller.

Reply:
Added bindings 3cafd24 and changed dtsi 6bf8fda.